bubble_sort.rb

```diff
@@ -0,0 +1,50 @@
+
+def bubble_sort
+	puts 'Enter number of items'
+
+	total_items = gets.chomp
+
+	items = total_items.to_i
+
+	array = []
+
+	0.upto(items - 1) do |i|
+		array[i] = gets.chomp
+	end
+
+	puts "Total Number of items in the array is:  #{total_items}"
+
+	puts "Youre array is #{array}"
+
+	array.each do
+		i = 0
+		count = 0
+		while i < items - 1
+			array[i], array[i + 1] = array[i + 1], array[i] if array[i] > array[i + 1]
+			i += 1
+			count += 1
+		end
+		break if count.zero?
+	end
+
+	puts "\nAnd sorted array is #{array}"
+end
+
+bubble_sort
+
+def bubble_sort_by(arr)
+	for i in 0..arr.length - 2 do
+		for j in 0..arr.length - 2 do
+			if (yield arr[j], arr[j + 1]).positive?
+				arr[j], arr[j + 1] = arr[j + 1], arr[j]
+			end
+		end
+	end
+	return arr
+end
```

Comment:
- [ ] Here, there is no need of adding the `return` statement. Kindly remove it. In Ruby, statements are implicitly return meaning that any statement in Ruby returns the value of the last evaluated expression without having to add the `return` statement to it.

Reply:
**Thank you so much for explaining in details! It helps me to understand the process and code at the same time**
- [x]    Its done now!